### ISSUE TITLE
[Web] Stuck styles depending on worklet (conditionals?)

### ISSUE THREAD

**Issue by brianreavis** [software-mansion/react-native-reanimated#2631]
## Description

This is hard to describe with words. I've found that using conditionals in worklets on Web can cause very odd behavior – particularly certain style properties getting "stuck". See [the snack](https://snack.expo.dev/@brianreavis/555fb7) for a walkthrough that reproduces the behavior. iOS and Android are not affected. Any ideas? I'm not sure that conditionals are specifically the issue; though it seems to be what causes this.

### Expected behavior

DOM should reflect `useAnimatedStyle` worklet result (no stuck properties)

### Actual behavior & steps to reproduce

(see snack)

## Snack or minimal code example

See the comments and uncomment Example A, B, C to walk through the odd behavior.
**https://snack.expo.dev/@brianreavis/555fb7**

## Package versions

2.2.0–2.2.4

## Affected platforms

- [ ] Android
- [ ] iOS
- [x] Web

**Comment by piaskowyk** [comment]
This is not a bug, because you animated both properties based on the same shared value, and according to your logic component was visible only if a component was placed at the top of the screen. If you want to achieve your goal you should use to shred value:
Example:
```js
import React, { useEffect } from 'react';
import { View, StyleSheet } from 'react-native';
import Animated, { useAnimatedStyle, useSharedValue, withTiming, withRepeat } from 'react-native-reanimated';

export default function App() {
  const progress = useSharedValue(0);
  const opacityProgress = useSharedValue(0);

  const style = useAnimatedStyle(() => {
    return {
      // opacity: opacityProgress.value > 0.1 ? 1 : 0,
      opacity: opacityProgress.value,
      transform: [{ translateY: progress.value * 100 }],
    };
  });

  useEffect(() => {
    opacityProgress.value = withRepeat(withTiming(1, { duration: 100 }), -1, true);
    progress.value = withRepeat(withTiming(1, { duration: 1000 }), -1, true);
  });

  return (
    <Animated.View style={[styles.container, style]} />
  );
}

const styles = StyleSheet.create({
  container: {
    width: 200,
    height: 200,
    backgroundColor: '#00ff00',
  },
});
```

https://user-images.githubusercontent.com/36106620/141975470-de606117-1f91-4458-9efd-da4a9aba4256.mov

**Comment by brianreavis** [comment]
Can we please reopen this? This _is_ a bug. My point was that Example B and C should have the _exact same_ flash frequency. In B and C, the opacity is set to:
```js
opacity: progress.value > 0.01 ? 1 : 0,
```
In practice, the flashing is wildly different due to a bug in reanimated that occurs as soon as the transform is added. Here's a screen recording with the transform really toned down to not be distracting:

**Example B:**
https://user-images.githubusercontent.com/118480/142045315-6eb7d844-b956-4e73-ae80-27f7806706c3.mov

**Example C:**
https://user-images.githubusercontent.com/118480/142045324-7f195c93-0749-467f-8dd5-42df2fd2349c.mov

> This is not a bug, because you animated both properties based on the same shared value

Why is this a problem? The issue is still present even if I drive the second property using a different shared value (see the [updated snack](https://snack.expo.dev/@brianreavis/555fb7)).

P.S. This example is a minimal contrived example to show behavior; I know there are more proper ways to flash things.

**Comment by brianreavis** [comment]
**Update:** This is caused by https://github.com/necolas/react-native-web/issues/1935 which is patched around in https://github.com/software-mansion/react-native-reanimated/pull/2280 (which was released in 2.2.4, but that release is broken by https://github.com/software-mansion/react-native-reanimated/issues/2635). We can leave this issue closed; when #2635 is fixed, the glitchy behavior should be resolved.
